packages/react-search-ui/src/SearchProvider.js

```diff
@@ -1,57 +1,61 @@
 import PropTypes from "prop-types";
 import React, { Component } from "react";
-import { SearchDriver } from "@elastic/search-ui";
 
+import { SearchDriver } from "@elastic/search-ui";
 import SearchContext from "./SearchContext";
 
 /**
- * The SearchProvider is the glue that connects the SearchDriver to
- * our React App.
- *
- * It "subscribes" to the SearchDriver in order to be notified of state
- * changes. It then syncs that state with its own state and passes that state
- * down to child components in a React Context. It will also pass down "Actions"
- * from the SearchDriver, which allow child components to update state.
- *
- * Any "Container" component placed within this context will have access
- * to the Driver's state and Actions.
- *
+ * The SearchProvider primarily holds a reference to the SearchDriver and
+ * exposes it to the rest of the application in a Context.
  */
 class SearchProvider extends Component {
   static propTypes = {
-    children: PropTypes.func.isRequired,
+    children: PropTypes.node.isRequired,
     // Not providing a shape here because the shape matches the shape of
     // SearchDriver. SearchDriver can do it's own parameter validation.
     config: PropTypes.object
   };
 
-  subscription = state => {
-    this.setState(state);
-  };
+  constructor() {
+    super();
+    this.state = {
+      driver: null
+    };
+  }
 
   componentDidMount() {
     const { config } = this.props;
-    this.driver = new SearchDriver(config);
-    this.setState(this.driver.getState());
-    this.driver.subscribeToStateChanges(this.subscription);
+    // This initialization is done inside of componentDidMount, because initializing the SearchDriver server side
```

Comment:
Just a nitpick:
I think it would be more concise and clear to check if the have `window` in the constructor:

```
const driver = typeof window !== 'undefined' ? new SearchDriver(config) : null;
this.setState({
      driver
    });
```

Reply:
@yakhinvadim I worry then, what would happen if we use this in SSR, like in Gatsby. Will the driver **ever** be initialized?

Reply:
We had a call with Jason and I agreed that we shouldn't risk not having driver initialized for slightly less code.